src/components/Login/LoginContainer.test.tsx

```diff
@@ -0,0 +1,308 @@
+import {
+  fireEvent,
+  render,
+  cleanup,
+  waitFor,
+  act,
+} from "@testing-library/react-native";
+import React from "react";
+import * as Linking from "expo-linking";
+import { InitialisationContainer } from "./LoginContainer";
+import { Sentry } from "../../utils/errorTracking";
+import { mockNavigation } from "../../test/helpers/navigation";
+import { CreateProvidersWrapper } from "../../test/helpers/providers";
+import { AlertModalContextProvider } from "../../context/alert";
+import * as auth from "../../services/auth";
+import "../../common/i18n/i18nMock";
+
+const key = "1e4457bc-f7d0-4329-a344-f0e3c75d8dd4";
+const endpoint = "https://somewhere.com";
+const invalidAuthMsg =
+  "Scan QR code again or get a new QR code from your in-charge.";
+const invalidOtpMsg = "Enter OTP again.";
+const invalidCountryCode = "Enter a valid country code.";
+const phoneNumber = "88888888";
+const countryCode = "+65";
+const sessionToken = "my-session-token";
+const otp = "000000";
+const expiry = new Date(2030, 0, 1);
+
+const barcodeScannerId = "barcode-scanner-camera";
+const buttonWithLoading = "dark-button-display-loading";
+const loginScanViewId = "login-scan-view";
+const loadingViewId = "loading-view";
+const loginMobileViewId = "login-mobile-number-view";
+const loginOTPViewId = "login-otp-card-view";
+const loginSendOTPButtonId = "login-send-otp-button";
+const loginSubmitOTPButtonId = "login-submit-otp-button";
+const scanButtonId = "base-button";
+const phoneInputId = "login-phone-number-input";
+const OTPInputId = "login-otp-input";
+const countryCodeInputId = "login-country-code-input";
+
+jest.mock("../../utils/errorTracking");
+const mockCaptureException = jest.fn();
+(Sentry.captureException as jest.Mock).mockImplementation(mockCaptureException);
+
+const mockDeepLink = jest.fn();
+jest.spyOn(Linking, "parseInitialURLAsync").mockImplementation(mockDeepLink);
+
+const mockRequestOTP = jest.spyOn(auth, "requestOTP");
+const mockValidateOTP = jest.spyOn(auth, "validateOTP");
+
+describe("LoginContainer", () => {
+  afterEach(() => {
+    mockCaptureException.mockReset();
+    mockDeepLink.mockReset();
+    mockRequestOTP.mockReset();
+    mockValidateOTP.mockReset();
+    cleanup();
+  });
+
+  it("should render LoginScanCard", () => {
+    expect.assertions(1);
+    const { getByTestId } = render(
+      <InitialisationContainer navigation={mockNavigation} />
+    );
+
+    const scanButton = getByTestId(scanButtonId);
+    fireEvent.press(scanButton);
+    expect(getByTestId(loginScanViewId)).not.toBeNull();
```

Comment:
noticed that regardless of whether we press the scanButton, `getByTestId(loginScanViewId)` will not be null